src/context/auth.jsx

```diff
@@ -49,25 +51,49 @@ const AuthProvider = ({ children }) => {
   };
 
   const handleRegister = async (email, password) => {
-    const res = await register(email, password);
-    if (res.data.error) {
-      setError(res.data.error);
-    } else {
-      setToken(res.data.token);
-      navigate("/verification");
-      setError("");
+    try {
+      const res = await register(email, password);
+      if (res.data.error) {
+        setError(res.data.error);
+      } else {
```

Comment:
same as above, if we add a `return` to the if statement, we don't need the `else`